tools/fork-network/src/cli.rs

```diff
@@ -0,0 +1,285 @@
+                .get_path(&store_path)
+                .expect("Migration snapshot must be enabled"),
+        );
+        let fork_snapshot_path = store_path.join("fork-snapshot");
+        std::fs::rename(&migration_snapshot_path, &fork_snapshot_path)?;
+
+        let storage = store_opener.open_in_mode(Mode::ReadWrite).unwrap();
+        let store = storage.get_hot_store();
+
+        let epoch_manager =
+            EpochManager::new_arc_handle(store.clone(), &near_config.genesis.config);
+        let runtime = NightshadeRuntime::from_config(
+            home_dir,
+            store.clone(),
+            &near_config,
+            epoch_manager.clone(),
+        );
+        let head = store.get_ser::<Tip>(DBCol::BlockMisc, HEAD_KEY)?.unwrap();
+        let shard_layout = epoch_manager.get_shard_layout(&head.epoch_id)?;
+        let all_shard_uids = shard_layout.get_shard_uids();
+
+        let mut flat_head: Option<CryptoHash> = None;
+        for shard_uid in all_shard_uids {
+            let flat_storage_status = store
+                .get_ser::<FlatStorageStatus>(DBCol::FlatStorageStatus, &shard_uid.to_bytes())?
+                .unwrap();
+            if let FlatStorageStatus::Ready(ready) = &flat_storage_status {
+                let flat_head_for_this_shard = ready.flat_head.hash;
+                if let Some(hash) = flat_head {
+                    if hash != flat_head_for_this_shard {
+                        panic!("Not all shards have the same flat head");
+                    }
+                }
+                flat_head = Some(flat_head_for_this_shard);
+            } else {
+                panic!(
+                    "Flat storage is not ready for shard {}: {:?}",
+                    shard_uid, flat_storage_status
+                );
+            }
+        }
+
+        let flat_head_block = store
+            .get_ser::<near_primitives::block::Block>(
+                DBCol::Block,
+                &flat_head.unwrap().try_to_vec().unwrap(),
+            )?
+            .unwrap();
+
+        let prev_state_roots = flat_head_block
+            .chunks()
+            .iter()
+            .map(|chunk| chunk.prev_state_root())
+            .collect::<Vec<_>>();
+
+        let self_validator = near_config.validator_signer.as_ref().unwrap();
+        let self_account = AccountInfo {
+            account_id: self_validator.validator_id().clone(),
+            amount: 50000000000000000000000000000_u128,
+            public_key: self_validator.public_key().clone(),
+        };
+
+        let mut storage_mutator = StorageMutator::new(
+            epoch_manager.clone(),
+            &*runtime,
+            flat_head_block.header().epoch_id(),
+            *flat_head_block.header().prev_hash(),
+            &prev_state_roots,
+        )?;
+
+        storage_mutator.set_account(
+            self_validator.validator_id().clone(),
+            Account::new(0, self_account.amount, CryptoHash::default(), 0),
+        )?;
+
```

Comment:
We can do this for multiple accounts right? In the current code it is just a question of convenience (it is easy to set up one validator). If I were to specify a JSON with all validators (ids, keys, amounts) it would be no problem to add them here too?
Also, for testing purposes we want some accounts to also have near, but not necessarily be validators (they need to generate traffic). It is also ok, right?

Reply:
Yeah, we can. This is a plumbing issue, I'm not too sure what the great way is to specify the input, so I just did this simple implementation. And yes, some test accounts with near will be needed, which we can also do. For that we also need to write the AccessKey nodes, but that's all the same thing.